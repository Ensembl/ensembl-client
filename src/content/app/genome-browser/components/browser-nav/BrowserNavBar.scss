@import 'src/styles/common';

.browserNavBar {
  display: flex;
  font-size: 12px;
  background: $light-grey;
  padding: 16px 26px;
  position: relative;
  z-index: 2;

  form {
    display: inline-block;
  }

  label {
    font-weight: 300;
    margin-right: 5px;
  }

  input {
    padding: 3px 9px;
  }
}

.errorText {
  color: $red;
}

.semiOpaque {
  opacity: 0.4;
}
