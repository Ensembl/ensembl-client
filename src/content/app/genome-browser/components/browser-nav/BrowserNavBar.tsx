/**
 * See the NOTICE file distributed with this work for additional information
 * regarding copyright ownership.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { memo } from 'react';
import { useSelector } from 'react-redux';
import classNames from 'classnames';

import BrowserNavBarControls from './BrowserNavBarControls';
import BrowserNavBarMain from './BrowserNavBarMain';

import { getIsTrackPanelOpened } from 'src/content/app/genome-browser/components/track-panel/state/trackPanelSelectors';

import styles from './BrowserNavBar.scss';

export const BrowserNavBar = () => {
  const expanded = !useSelector(getIsTrackPanelOpened);

  const className = classNames(styles.browserNavBar, {
    [styles.browserNavBarExpanded]: expanded
  });

  return (
    <div className={className}>
      <BrowserNavBarControls />
      <BrowserNavBarMain />
    </div>
  );
};

export default memo(BrowserNavBar);
