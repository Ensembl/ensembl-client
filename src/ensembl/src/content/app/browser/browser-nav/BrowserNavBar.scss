@import 'src/styles/common';

.browserNavBar {
  @include flex;
  background: $ens-light-grey;
  padding: 16px;

  @include breakpoint(large) {
    width: calc(100vw - 356px);
  }

  &.browserNavBarExpanded {
    width: 100vw;
  }
}
