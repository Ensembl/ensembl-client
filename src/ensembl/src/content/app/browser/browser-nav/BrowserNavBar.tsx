import React, { FunctionComponent, RefObject } from 'react';
import { connect } from 'react-redux';

import { browserNavConfig, BrowserNavItem } from '../browserConfig';

import { RootState } from 'src/store';
import { getBrowserNavStates } from '../browserSelectors';
import { BrowserNavStates } from '../browserState';

import BrowserNavIcon from './BrowserNavIcon';

import styles from './BrowserNavBar.scss';

type StateProps = {
  browserNavStates: BrowserNavStates;
};

type DispatchProps = {};

type OwnProps = {
  browserRef: RefObject<HTMLDivElement>;
};

type BrowserNavBarProps = StateProps & DispatchProps & OwnProps;

export const BrowserNavBar: FunctionComponent<BrowserNavBarProps> = (
  props: BrowserNavBarProps
) => {
  const browserImageEl = props.browserRef.current as HTMLDivElement;

  return (
    <div className={styles.browserNavBar}>
      <dl>
        {browserNavConfig.map((item: BrowserNavItem, index: number) => (
          <BrowserNavIcon
            key={item.name}
            browserNavItem={item}
            browserImageEl={browserImageEl}
            maxState={props.browserNavStates[index]}
          />
        ))}
      </dl>
    </div>
  );
};

const mapStateToProps = (state: RootState): StateProps => ({
  browserNavStates: getBrowserNavStates(state)
});

const mapDispatchToProps: DispatchProps = {};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BrowserNavBar);
