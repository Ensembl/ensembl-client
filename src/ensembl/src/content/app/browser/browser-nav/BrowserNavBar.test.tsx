import React from 'react';
import { render } from 'enzyme';

jest.mock('./BrowserNavBarControls', () => () => (
  <div>BrowserNavBarControls</div>
));
jest.mock('./BrowserNavBarMain', () => () => <div>BrowserNavBarControls</div>);

import { BrowserNavBar } from './BrowserNavBar';

describe('<BrowserNavBar />', () => {
  describe('rendering', () => {
    it('correctly interprets the "expanded" prop', () => {
      const contractedBar = render(<BrowserNavBar expanded={false} />);
      expect(contractedBar.hasClass('browserNavBarExpanded')).toBe(false);

      const expandedBar = render(<BrowserNavBar expanded={true} />);
      expect(expandedBar.hasClass('browserNavBarExpanded')).toBe(true);
    });
  });
});
