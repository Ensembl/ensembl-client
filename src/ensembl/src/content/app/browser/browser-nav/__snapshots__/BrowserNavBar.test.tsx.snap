// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowserNavBar /> renders correctly 1`] = `
<div
  className="browserNavBar"
>
  <dl>
    <React.Memo
      browserNavItem={
        Object {
          "description": "navigate up",
          "detail": Object {
            "move_up_px": 50,
          },
          "icon": Object {
            "off": Object {},
            "on": Object {},
          },
          "name": "navigate-up",
        }
      }
      key="navigate-up"
    />
    <React.Memo
      browserNavItem={
        Object {
          "description": "navigate down",
          "detail": Object {
            "move_down_px": 50,
          },
          "icon": Object {
            "off": Object {},
            "on": Object {},
          },
          "name": "navigate-down",
        }
      }
      key="navigate-down"
    />
    <React.Memo
      browserNavItem={
        Object {
          "description": "zoom out",
          "detail": Object {
            "zoom_by": -0.3,
          },
          "icon": Object {
            "off": Object {},
            "on": Object {},
          },
          "name": "zoom-out",
        }
      }
      key="zoom-out"
    />
    <React.Memo
      browserNavItem={
        Object {
          "description": "zoom in",
          "detail": Object {
            "zoom_by": 0.3,
          },
          "icon": Object {
            "off": Object {},
            "on": Object {},
          },
          "name": "zoom-in",
        }
      }
      key="zoom-in"
    />
    <React.Memo
      browserNavItem={
        Object {
          "description": "navigate left",
          "detail": Object {
            "move_left_px": 50,
          },
          "icon": Object {
            "off": Object {},
            "on": Object {},
          },
          "name": "navigate left",
        }
      }
      key="navigate left"
    />
    <React.Memo
      browserNavItem={
        Object {
          "description": "navigate right",
          "detail": Object {
            "move_right_px": 50,
          },
          "icon": Object {
            "off": Object {},
            "on": Object {},
          },
          "name": "navigate right",
        }
      }
      key="navigate right"
    />
  </dl>
</div>
`;
